Mathlib/Combinatorics/SimpleGraph/Path.lean

```diff
@@ -961,6 +961,24 @@ lemma mem_coe_supp_of_adj {v w : V} {H : Subgraph G} {c : ConnectedComponent H.c
   rw [← (mem_supp_iff _ _).mp h.1]
   exact ⟨connectedComponentMk_eq_of_adj <| Subgraph.Adj.coe <| h.2 ▸ hadj.symm, rfl⟩
 
+lemma connectedComponentMk_supp_subset_supp {G'} {v : V} (h : G ≤ G') (c' : G'.ConnectedComponent)
+    (hc' : v ∈ c'.supp) : (G.connectedComponentMk v).supp ⊆ c'.supp := by
+  intro v' hv'
+  simp only [mem_supp_iff, ConnectedComponent.eq] at hv' ⊢
+  rw [ConnectedComponent.sound (hv'.mono h)]
+  exact hc'
+
+lemma union_supp_eq_supp {G G' : SimpleGraph V} (h : G ≤ G') (c' : ConnectedComponent G') :
+    ⋃ c : {c : ConnectedComponent G | c.supp ⊆ c'.supp}, c.1.supp = c'.supp := by
```

Comment:
Maybe
```suggestion
lemma biUnion_supp_eq_supp {G G' : SimpleGraph V} (h : G ≤ G') (c' : ConnectedComponent G') :
    ⋃ (c : ConnectedComponent G) (hc : c.supp ⊆ c'.supp}, c.1.supp = c'.supp := by
```

Reply:
I tried this, two remarks:

- Is biUnion the right name for this? This is just a double union, whereas the biUnion is over a set, right?
- I'm having trouble using the lemma because the motive is incorrect. Previously I managed to fix this by adding the `DecidablePred`, but I don't really know how to get there now.

Reply:
Managed to get this to work, but I'm guessing we want to do some golfing/extraction before merging. In particular the argument to Finset.filter_congr seems a prime candidate. Any thoughts what we would want to do there?

Reply:
> * Is biUnion the right name for this? This is just a double union, whereas the biUnion is over a set, right?

`biUnion` is precisely the special case of `iUnion₂` where the inner `iUnion` runs over a Prop